src/core/service.c

```diff
@@ -1562,21 +1650,38 @@ static int service_spawn(
         }
 
         if (flags & EXEC_SETENV_RESULT) {
-                if (asprintf(our_env + n_env++, "SERVICE_RESULT=%s", service_result_to_string(s->result)) < 0)
+                r = asprintf(our_env + n_env, "SERVICE_RESULT=%s", service_result_to_string(s->result));
```

Comment:
are the changes to this branch of the if block necessary? seems to be just some reshuffling?

Reply:
It's worse, I accidently changed the conditions here. I've completely reverted this section.